src/coreclr/tools/aot/ILCompiler.ReadyToRun/Compiler/DependencyAnalysis/ReadyToRun/DelegateCtorSignature.cs

```diff
@@ -47,7 +47,7 @@ public override ObjectData GetData(NodeFactory factory, bool relocsOnly = false)
                     enforceDefEncoding: false,
                     enforceOwningType: false,
                     innerContext,
-                    isInstantiatingStub: _targetMethod.Method.HasInstantiation);
+                    isInstantiatingStub: _targetMethod.Method.HasInstantiation || _targetMethod.Method.OwningType.HasInstantiation || _methodToken.ConstrainedType != null);
```

Comment:
Is this change only necessary for SVMs? If so we should only add the extra checks for static methods, which are virtual in some way.

Reply:
Good point, modified in the latest commit.